### PR TITLE
SWC-6070 - Clear react-query entity cache on updated event

### PR DESCRIPTION
**Pull request by nickgros**
When we fire an EntityUpdatedEvent, we should clear the entity-related data cached by react-query.

### CODE REVIEW

File: src/main/java/org/sagebionetworks/web/client/events/EntityUpdatedEvent.java [hunk 1]
```diff
@@ -2,5 +2,26 @@
...
 public class EntityUpdatedEvent extends GenericEvent {
+	private String entityId = null;
+
+	public EntityUpdatedEvent() {
+	}
+
+
+	public EntityUpdatedEvent(String entityId) {
+		this.entityId = entityId;
+	}
```
Comment (nickgros) [inline src/main/java/org/sagebionetworks/web/client/events/EntityUpdatedEvent.java]: Optionally supply an entity ID when creating this event

File: src/main/java/org/sagebionetworks/web/client/presenter/EntityPresenter.java [hunk 2]
```diff
@@ -268,6 +273,9 @@ public Widget asWidget() {
 
 	@EventHandler
 	public void onEntityUpdatedEvent(EntityUpdatedEvent event) {
+		SynapseReactClientQueryKey queryKey = SynapseReactClientQueryKey.create("entity", event.getEntityId());
+		// The query key is an array where the first element is the object we constructed
+		queryClient.invalidateQueries(new Object[]{queryKey});
```
Comment (nickgros) [inline src/main/java/org/sagebionetworks/web/client/presenter/EntityPresenter.java:278]: Invalidate the react-query client for the updated entity, or all entities if the event's entity ID is null

File: src/main/java/org/sagebionetworks/web/client/events/EntityUpdatedEvent.java [hunk 3]
```diff
@@ -2,5 +2,28 @@
...
+
+	public EntityUpdatedEvent() {
+		super();
+	}
+
+
+	public EntityUpdatedEvent(String entityId) {
+		super();
+		this.entityId = entityId;
+	}
```
Comment (nickgros) [inline src/main/java/org/sagebionetworks/web/client/events/EntityUpdatedEvent.java:20]: Optionally supply an entity ID when creating this event.

**Review by jay-hodgson — APPROVED** [review]

**Comment by jay-hodgson** [comment]
Merged to include in Synapse stack 403

**Pull request merged**